lib/restforce/sobject.rb

```diff
@@ -16,6 +16,12 @@ def describe_layouts(layout_id = nil)
       @client.describe_layouts(sobject_type, layout_id)
     end
 
+    def refresh
+      ensure_id
+      new_record = @client.find(sobject_type, self.Id)
+      deep_update(new_record.attrs)
```

Comment:
Why do you use `deep_update` rather than just returning the result of `#find`?

Reply:
It allows for:

```
order.refresh
```

Without having to reassign:

```
order = order.refresh
```

Seemed a bit cleaner IMHO, but returning the new object as well is probably better:

```suggestion
      deep_update(new_record.attrs)
      new_record
```

Reply:
Excellent point! That makes sense - it gives you ActiveRecord-like behaviour.